### ISSUE TITLE
Initial sync is slow

### ISSUE THREAD

**Issue by vurpo** [mujx/nheko#86]
The first sync that nheko does after you login takes a long time to complete, and on some servers this is even so long that it causes a timeout (see #19).

I did an initial sync in cURL using the exact parameters and filters that nheko uses, and the ones that Riot-web uses, to try to figure out the cause of this. These are the results:

Riot-web:
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 22.9M  100 22.9M    0     0   353k      0  0:01:06  0:01:06 --:--:--  800k
```

nheko:
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 58.3M  100 58.3M    0     0   215k      0  0:04:37  0:04:37 --:--:-- 1998k
```

A very clear difference in the size of the server's reply, and the time the request took to finish. Riot and nheko use basically equivalent filters for this request (nheko just filters out ephemeral events and presence, which it doesn't currently use, but the filters are otherwise identical).

Riot-web doesn't put any extra parameters in the request (other than a `_cacheBuster` parameter, which I don't know the purpose of). Nheko, on the other hand, uses `full_state=true` and `set_presence=online`. Personally I believe the `full_state` is the biggest culprit here.

I haven't done any experiments beyond this, but I will try to see if I can fix this in nheko in the coming days.

**Comment by mujx** [comment]
I did a comparison between those two on my matrix.org account and they're exactly the same. With or without `full_state=true`. Did you test on your own homeserver?

**Comment by vurpo** [comment]
I did these tests with my own account on my own homeserver.

**Comment by vurpo** [comment]
I don't know exactly what would cause this difference, but the only differences between what Riot does and what nheko does is that nheko has the ephemeral and presence filters that Riot doesn't have, and nheko adds `full_state=true` and `set_presence=online`. I also did the tests a couple of times to see if there was caching affecting the results, but I got the exact same results every time.

Usually when nheko is logging in to my account, my HS also stops responding (or responds too slowly) to other clients trying to actively sync, causing them to claim the connection has been lost. This also happened when I did these sync tests using curl.
